bundles/org.openhab.binding.openweathermap/src/main/java/org/openhab/binding/openweathermap/internal/connection/OpenWeatherMapConnection.java

```diff
@@ -314,11 +314,10 @@ private Map<String, String> getRequestParams(OpenWeatherMapAPIConfiguration conf
 
         Map<String, String> params = new HashMap<>();
         // API key (see https://openweathermap.org/appid)
-        String apikey = config.apikey;
-        if (apikey == null || (apikey = apikey.trim()).isEmpty()) {
+        if (config.apikey.isBlank()) {
             throw new ConfigurationException("@text/offline.conf-error-missing-apikey");
         }
-        params.put(PARAM_APPID, apikey);
+        params.put(PARAM_APPID, config.apikey);
```

Comment:
Just a short note that `apikey` is no longer trimmed, so this could break configurations having whitespaces at the beginning or end of the key. However, I think it makes sense to be more strict.

Reply:
Yes, apikey should not have whitespaces. Intentional fix.